pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/ServiceBusProcessor.kt

```diff
@@ -96,92 +104,164 @@ class ServiceBusProcessor {
-     * @param value Any
-     */
-    private fun isNullOrEmpty(value: Any?): Boolean {
-        return when (value) {
-            null -> true
-            is String -> value.isEmpty()
-            is Collection<*> -> value.isEmpty()
-            is Map<*, *> -> value.isEmpty()
-            else -> false // You can adjust this to your needs
+        } catch (e: Exception) {
+            LOGGER.error("Report rejected: Malformed JSON or error processing the report - ${e.message}")
+            throw e
         }
     }
 
-    /**
-     * Convert the malformed content to base64 encoded string
-     * @param obj Any
-     */
-    private fun convertToStringOrBase64(obj: Any?): String {
-        val bytes = when (obj) {
-            is ByteArray -> obj
-            else -> obj.toString().toByteArray()
+    companion object {
+        /**
+         *  Function to send the invalid data reasons to the deadLetter queue
+         *  @param invalidData MutableList<String>
+         *  @param createReportMessage CreateReportSBMessage
+         */
+        @JvmStatic
+        private fun SendToDeadLetter(invalidData:MutableList<String>, createReportMessage: CreateReportSBMessage){
+            if (invalidData.isNotEmpty()) {
+                //This should not run for unit tests
+                if (System.getProperty("isTestEnvironment") != "true") {
+                    // Write the content of the dead-letter reports to CosmosDb
+                    ReportManager().createDeadLetterReport(
+                        createReportMessage.uploadId,
+                        createReportMessage.dataStreamId,
+                        createReportMessage.dataStreamRoute,
+                        createReportMessage.stageName,
+                        createReportMessage.dispositionType,
+                        createReportMessage.contentType,
+                        createReportMessage.content,
+                        invalidData
+                    )
+                }
+                throw BadRequestException(invalidData.joinToString(separator = ","))
+            }
+        }
+        /**
+         *  Function to process the error by logging it and adding to the invalidData list and sending it to deadletter
+         *  @param reason String
+         *  @param invalidData MutableList<String>
+         *  @param createReportMessage CreateReportSBMessage
+         */
+        @JvmStatic
+        private fun ProcessError(reason:String, invalidData:MutableList<String>, createReportMessage: CreateReportSBMessage) {
+            LOGGER.error(reason)
+            invalidData.add(reason)
+            SendToDeadLetter(invalidData, createReportMessage)
+        }
+
+        /**
+         *  Function to validate the schema based on the schema file and the json contents passed into it
+         *  @param schemaFile String
+         *  @param objectMapper ObjectMapper
+
+         */
+        @JvmStatic
```

Comment:
I don't think we need to declare this @JvmStatic as it will only be used by Kotlin code.